garage/tf/policies/gaussian_mlp_policy_with_model.py

```diff
@@ -77,37 +78,42 @@ def __init__(self,
             std_output_nonlinearity=std_output_nonlinearity,
             std_parameterization=std_parameterization)
 
-        _, self._mean_var, self._log_std_var, _, self._dist = (
-            self.model.build(state_input))
-
-        self._f_dist = lambda x: tf.get_default_session().run(
-            [self._mean_var, self._log_std_var], feed_dict={state_input: x})
+        self.model.build(state_input)
 
     @property
     def vectorized(self):
         """Vectorized or not."""
         return True
 
-    def dist_info_sym(self,
-                      obs_var,
-                      state_info_vars=None,
-                      name='dist_info_sym'):
+    def dist_info_sym(self, obs_var, state_info_vars=None, name='default'):
         """Symbolic graph of the distribution."""
         _, mean_var, log_std_var, _, _ = self.model.build(obs_var, name=name)
         return dict(mean=mean_var, log_std=log_std_var)
 
     def get_action(self, observation):
         """Get action from the policy."""
         flat_obs = self.observation_space.flatten(observation)
-        mean, log_std = [x[0] for x in self._f_dist([flat_obs])]
+        _f_dist = tensor_utils.compile_function(
```

Comment:
i think at this point it would be clearer to define an actual method _f_dist which just has the tf.get_default_session.run() call in it.